src/wd.rs

```diff
@@ -373,3 +372,440 @@ impl NewSessionResponse {
+    pub fn orientation(mut self, orientation: PrintOrientation) -> Self {
+        self.orientation = orientation;
+
+        self
+    }
+
+    /// Sets the scale of the printed page.
+    ///
+    /// Default: 1.
+    pub fn scale(mut self, scale: f64) -> Self {
+        self.scale = scale;
+
+        self
+    }
+
+    /// Sets whether or not to print the backgrounds of the page.
+    ///
+    /// Default: false.
+    pub fn background(mut self, background: bool) -> Self {
+        self.background = background;
+
+        self
+    }
+
+    /// Sets the size of the printed page.
+    ///
+    /// Default: [`PrintSize::A4`].
+    pub fn size(mut self, size: PrintSize) -> Self {
+        self.size = size;
+
+        self
+    }
+
+    /// Sets the margins of the printed page.
+    ///
+    /// Default: `1x1x1x1 cm`.
+    pub fn margins(mut self, margins: PrintMargins) -> Self {
+        self.margins = margins;
+
+        self
+    }
+
+    /// Sets ranges of pages to print.
+    ///
+    /// An empty `ranges` prints all pages, which is the default.
+    pub fn page_ranges(mut self, ranges: Vec<PrintPageRange>) -> Self {
+        self.page_ranges = ranges;
+
+        self
+    }
+
+    /// Sets whether or not to resize the content to fit the page width,
+    /// overriding any page width specified in the content of pages to print.
+    ///
+    /// Default: true.
+    pub fn shrink_to_fit(mut self, shrink_to_fit: bool) -> Self {
+        self.shrink_to_fit = shrink_to_fit;
+
+        self
+    }
+}
+
+/// The print configuration.
+#[derive(Debug, Clone, PartialEq)]
+pub struct PrintConfiguration {
+    orientation: PrintOrientation,
+    scale: f64,
+    background: bool,
+    size: PrintSize,
+    margins: PrintMargins,
+    page_ranges: Vec<PrintPageRange>,
+    shrink_to_fit: bool,
+}
+
+impl PrintConfiguration {
+    /// Creates a [`PrintConfigurationBuilder`] to configure a [`PrintConfiguration`].
+    pub fn builder() -> PrintConfigurationBuilder {
+        PrintConfigurationBuilder::default()
+    }
+
+    pub(crate) fn into_params(self) -> PrintParameters {
+        PrintParameters {
+            orientation: self.orientation.into_params(),
+            scale: self.scale,
+            background: self.background,
+            page: self.size.into_params(),
+            margin: self.margins.into_params(),
+            page_ranges: self
+                .page_ranges
+                .into_iter()
+                .map(|page_range| page_range.into_params())
+                .collect(),
+            shrink_to_fit: self.shrink_to_fit,
+        }
+    }
+}
+
+impl Default for PrintConfiguration {
+    fn default() -> Self {
+        PrintConfiguration {
+            orientation: PrintOrientation::default(),
+            scale: 1.0,
+            background: false,
+            size: PrintSize::default(),
+            margins: PrintMargins::default(),
+            page_ranges: Vec::new(),
+            shrink_to_fit: true,
+        }
```

Comment:
Instead of repeating all the defaults here, I think we can just

```suggestion
        PrintConfigurationBuilder::default().build().expect("default configuration is buildable")
```